static/js/controllers/edit-user.js

```diff
@@ -101,13 +108,16 @@
       };
 
       var validateName = function() {
-        if (!$scope.editUserModel.name) {
-          $scope.errors.name = $translate.instant('field is required', {
-            field: $translate.instant('User Name')
-          });
-          return false;
+        return validateRequired('name', 'User Name');
+      };
+
+      var validateRestrictedUser = function() {
```

Comment:
I'm slightly uncomfortable that `validateRestrictedUser()` returns `true` if the user is _not_ restricted.